### PR TITLE
Add minimum comment count to enable custom comment depth

### LINKED ISSUE #3456: Option to only apply comment depth if # comments > threshold
threshold default of .. 50?
customizable per subreddit
option (enabled) to always apply depth if # comments not immediately available

### PR DESCRIPTION
**Pull request by dhensche**
Resolves #3456

### CODE REVIEW

File: lib/modules/commentDepth.js [hunk 1]
```diff
@@ -29,7 +34,7 @@ module.options = {
 		value: false,
 		description: 'Set depth on links to particular comments with context.',
 	},
-	subredditCommentDepths: {
+	subredditCommentDepthConfigurations: {
```
Comment (dhensche) [inline lib/modules/commentDepth.js]: Totally forgot about possible option migration issues when doing this rename

File: lib/modules/commentDepth.js [hunk 2]
```diff
@@ -39,9 +44,14 @@ module.options = {
 		}, {
 			name: 'commentDepth',
 			type: 'text',
+			value: '4',
```
Comment (dhensche) [inline lib/modules/commentDepth.js]: yeah, I can update all the fields to be a plain number if desired

Comment (erikdesjardins) [inline lib/modules/commentDepth.js]: I think all text option default values should be strings (until we have a number type) to emphasize that you should use `parseInt`/`parseFloat` since it _could_ be saved as a string

File: lib/modules/commentDepth.js [hunk 3]
```diff
@@ -39,9 +44,14 @@ module.options = {
...
+		}, {
+			name: 'minimumComments',
+			type: 'text',
+			value: '50',
```
Comment (erikdesjardins) [inline lib/modules/commentDepth.js]: In a larger module I'd say no because it might be easy to miss, but this module is so small that it's not a big deal

Comment (andytuba) [inline lib/modules/commentDepth.js]: Yeah, it's a smallish module and I don't expect it to scope creep too much.

As an academic exercise, I would implement it like:

```javascript
module.loadDynamicOptions = function() {
    module.options.subredditDepthsWhateverItsCalled.fields[0].value = module.options.defaultDepth.value;
    module.options.subredditDepthsWhateverItsCalled.fields[1].value = module.options.defaultMinimum.value;
};

```

**Comment by andytuba** [comment]
Somebody give it a spin and I'd say merge in?

**Comment by dhensche** [comment]
@erikdesjardins left a comment about leaving the numeric text option default values as strings right before I committed the change to plain numbers. @andytuba should I revert those back to strings before merging in?

**Comment by andytuba** [comment]
ah yeah, let's go back to strings, sorry for the waffling!
